components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -698,6 +702,14 @@ void AdsServiceImpl::InitializeNotificationAdsPrefChangeRegistrar() {
       base::BindRepeating(&AdsServiceImpl::NotifyPrefChanged,
                           base::Unretained(this),
                           prefs::kMaximumNotificationAdsPerHour));
+  auto position_callback = base::BindRepeating(
+      &AdsServiceImpl::OnNotificationPositionChanged, base::Unretained(this));
+  pref_change_registrar_.Add(
```

Comment:
This is not a good place to collect p3a metrics because these settings are changed multiple times when the notification popup is dragged. The better place is `AdsServiceImpl::ShowNotificationAd` when `ShouldShowCustomNotificationAds()` is true.

Reply:
thanks for the pointer, but i think that might make QA difficult since one would have to wait for another notification to popup before the metric updates. i decided to add a debouncing mechanism instead.

Reply:
Could you please explain when the metrics is sent? Seems like it is not connected with notification ad serving, but to dragging. It is also sent on browser start. But it could happen that user doesn't get any notifications between browser restarts.

Reply:
When notification is dragged then preference is changed multiple times with different coordinates. Which one is used for the metrics?

Reply:
The metric is reported every week for users that have a custom notification position configured. It doesn't really matter how often or when it gets reported internally, the P3A reporter will simply upload the most up-to-date value every week.

> When notification is dragged then preference is changed multiple times with different coordinates. Which one is used for the metrics?

The last update for both preferences, hence the debouncing mechanism.

Reply:
hmm, how we report that custom notification is in default position then?

Reply:
I have a concern that the metric is connected to browser restarts but not to custom notification ad viewed event. We will have this metric even if user don't see any ads. Also if user doesn't restart their browser then we will get less metrics.

Reply:
> hmm, how we report that custom notification is in default position then?

we would simply not report, because the user doesn't have a custom position configured

> We will have this metric even if user don't see any ads.

This is okay, the wording of the question is: "If you have a custom position configured for ads notifications, what area is it in?"

It is not: "If you have a custom position configured for ads notifications and you viewed one ad, what area is it in?"

> Also if user doesn't restart their browser then we will get less metrics

the browser will restart when updating to a new version which includes the metric.

Reply:
basing on release schedule there could be a month between versions: https://github.com/brave/brave-browser/wiki/Brave-Release-Schedule

Reply:
> But maybe it is unlikely that user doesn't restart their browser at all, so it should be fine?

as long as the user updates periodically, we should be fine